From: Nithin Sujir <nsujir@broadcom.com>
Date: Thu, 23 May 2013 11:11:27 +0000
Subject: [PATCH 046/108] tg3: Remove unnecessary lock around tg3_flag_set
Patch-mainline: v3.11-rc1
Git-commit: f2068b80ca59d63ca21ee1da829c9fe09f25b08e
References: bsc#908458 FATE#317507


The spinlock was needed when flags used to be a u32 and set/cleared
using bit operations. Now that we use the atomic set_bit, this lock
isn't needed.

Signed-off-by: Nithin Nayak Sujir <nsujir@broadcom.com>
Signed-off-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/broadcom/tg3.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -6309,9 +6309,7 @@ static void tg3_tx_recover(struct tg3 *t
 		    "Please report the problem to the driver maintainer "
 		    "and include system chipset information.\n");
 
-	spin_lock(&tp->lock);
 	tg3_flag_set(tp, TX_RECOVERY_PENDING);
-	spin_unlock(&tp->lock);
 }
 
 static inline u32 tg3_tx_avail(struct tg3_napi *tnapi)
